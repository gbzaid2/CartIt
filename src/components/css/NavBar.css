#navbar {
  text-align: left;
  margin: 4px;
  text-align: center;
}

div.navbar-brand {
  color: white;
  font-size: 20px;
  text-decoration: underline;
}

#burger {
  color: white;
}

.link {
  color: black;
}

#searchbar {
  border-color: #525252;
  color: white;
}

input:focus {
  outline: none;
}
