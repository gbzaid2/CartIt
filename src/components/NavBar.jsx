import React, { useEffect, useRef } from "react";
import { Link } from "react-router-dom";
import "./css/NavBar.css";

const NavBar = (props) => {
  const node = useRef();
  useEffect(() => {
    document.addEventListener("mousedown", handleMoveOutside);
    document.addEventListener("keydown", handleMoveOutside);
    return () => {
      document.removeEventListener("mousedown", handleMoveOutside);
      document.removeEventListener("keydown", handleMoveOutside);
    };
  }, []);
  const toggleNavBar = () => {
    let navToggle = document.querySelector(".nav-toggle");
    let navbarMenu = document.querySelector(".navbar-menu");
    navToggle.classList.toggle("is-active");
    navbarMenu.classList.toggle("is-active");
  };
  const handleMoveOutside = (event) => {
    if (node.current.contains(event.target)) {
      return;
    } else {
      let navToggle = document.querySelector(".nav-toggle");
      let navbarMenu = document.querySelector(".navbar-menu");
      navToggle.classList.remove("is-active");
      navbarMenu.classList.remove("is-active");
    }
  };
  return (
    <nav id="navbar" className="navbar is-mobile is-spaced is-clearfix" ref={node}>
      <div className="navbar-brand">
        <div className="navbar-item" id="brand">
          <strong>
            <Link className="link" to="/">
              Greenfield
            </Link>
          </strong>
        </div>
        <div className="navbar-burger nav-toggle" onClick={toggleNavBar}>
          <span></span>
          <span></span>
          <span></span>
        </div>
      </div>
      <div className="navbar-menu">
        <div className="navbar-start">
          <div className="navbar-item is-hoverable" onClick={toggleNavBar}>
            <Link className="link" to="/">
              Home
            </Link>
          </div>
          <div className="navbar-item is-hoverable" onClick={toggleNavBar}>
            <Link className="link" to="/example-list">
              Example List
            </Link>
          </div>
          <div className="navbar-item is-hoverable" onClick={toggleNavBar}>
            <Link className="link" to="/example-form">
              Example Form
            </Link>
          </div>
        </div>
      </div>
    </nav>
  );
};

export default NavBar;
