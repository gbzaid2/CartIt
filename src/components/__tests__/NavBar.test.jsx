import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import NavBar from "../../components/NavBar.jsx";
import { MemoryRouter as Router } from "react-router-dom";

test("NavBar renders properly", () => {
  const wrapper = shallow(<NavBar />);
  const navbar = wrapper.find(".navbar");
  expect(navbar.exists()).toBe(true);
});

test("NavBar renders a snapshot properly", () => {
  const tree = renderer
    .create(
      <Router location="/">
        <NavBar />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
