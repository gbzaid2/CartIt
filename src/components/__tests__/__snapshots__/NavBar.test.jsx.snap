// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders a snapshot properly 1`] = `
<nav
  className="navbar is-mobile is-spaced is-clearfix"
  id="navbar"
>
  <div
    className="navbar-brand"
  >
    <div
      className="navbar-item"
      id="brand"
    >
      <strong>
        <a
          className="link"
          href="/"
          onClick={[Function]}
        >
          Greenfield
        </a>
      </strong>
    </div>
    <div
      className="navbar-burger nav-toggle"
      onClick={[Function]}
    >
      <span />
      <span />
      <span />
    </div>
  </div>
  <div
    className="navbar-menu"
  >
    <div
      className="navbar-start"
    >
      <div
        className="navbar-item is-hoverable"
        onClick={[Function]}
      >
        <a
          className="link"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </div>
      <div
        className="navbar-item is-hoverable"
        onClick={[Function]}
      >
        <a
          className="link"
          href="/example-list"
          onClick={[Function]}
        >
          Example List
        </a>
      </div>
      <div
        className="navbar-item is-hoverable"
        onClick={[Function]}
      >
        <a
          className="link"
          href="/example-form"
          onClick={[Function]}
        >
          Example Form
        </a>
      </div>
    </div>
  </div>
</nav>
`;
